examples/nas/oneshot/spos/search.py

```diff
@@ -37,49 +108,70 @@ def __call__(self, ir_model):
-                                val_dataloaders=pl.DataLoader(test_dataset, batch_size=64),
-                                max_epochs=2, gpus=1)
-
-    simple_strategy = strategy.RegularizedEvolution(model_filter=LatencyFilter(threshold=100, predictor=base_predictor),
-                                                    sample_size=1, population_size=2, cycles=2)
-    exp = RetiariiExperiment(base_model, trainer, strategy=simple_strategy)
+def _main():
+    parser = argparse.ArgumentParser("SPOS Evolutional Search")
+    parser.add_argument("--imagenet-dir", type=str, default="./data/imagenet")
+    parser.add_argument("--checkpoint", type=str, default="./data/checkpoint-150000.pth.tar")
+    parser.add_argument("--spos-preprocessing", action="store_true", default=False,
+                        help="When true, image values will range from 0 to 255 and use BGR "
+                             "(as in original repo).")
+    parser.add_argument("--seed", type=int, default=42)
+    parser.add_argument("--workers", type=int, default=6)
+    parser.add_argument("--train-batch-size", type=int, default=128)
+    parser.add_argument("--train-iters", type=int, default=200)
+    parser.add_argument("--test-batch-size", type=int, default=512)
+    parser.add_argument("--log-frequency", type=int, default=10)
+    parser.add_argument("--label-smoothing", type=float, default=0.1)
+
+    args = parser.parse_args()
+
+    # use a fixed set of image will improve the performance
+    torch.manual_seed(args.seed)
+    torch.cuda.manual_seed_all(args.seed)
+    np.random.seed(args.seed)
+    random.seed(args.seed)
+    torch.backends.cudnn.deterministic = True
+
+    assert torch.cuda.is_available()
+
+    base_model = ShuffleNetV2OneShot()
+    base_latency_predictor = 'cortexA76cpu_tflite21'
+    criterion = CrossEntropyLabelSmooth(1000, args.label_smoothing)
+    if args.spos_preprocessing:
+        trans = nni.trace(transforms.Compose)([
+            nni.trace(transforms.RandomResizedCrop)(224),
+            nni.trace(transforms.ColorJitter)(brightness=0.4, contrast=0.4, saturation=0.4),
+            nni.trace(transforms.RandomHorizontalFlip)(0.5),
+            nni.trace(ToBGRTensor)(),
+        ])
+    else:
+        trans = nni.trace(transforms.Compose)([
+            nni.trace(transforms.RandomResizedCrop)(224),
+            nni.trace(transforms.ToTensor)()
+        ])
+    train_dataset = nni.trace(datasets.ImageNet)(args.imagenet_dir, split='train', transform=trans)
+    val_dataset = nni.trace(datasets.ImageNet)(args.imagenet_dir, split='val', transform=trans)
+
+    evaluator = FunctionalEvaluator(evaluate_acc, criterion=criterion, args=args, train_dataset=train_dataset, val_dataset=val_dataset)
+    evolution_strategy = strategy.RegularizedEvolution(
+        # model_filter=LatencyFilter(threshold=100, predictor=base_latency_predictor),
+        sample_size=1, population_size=2, cycles=2)
```

Comment:
You can make these hyper-parameters appear in argparse.

By default, I think `sample_size=10`, `population_size=50`, `cycles=10`.